composer/datasets/streaming/format.py

```diff
@@ -178,12 +178,15 @@ def dumps(self) -> bytes:
         Returns:
             bytes: The serialized form.
         """
-        header = np.array([self.total_samples, self.total_bytes, self.num_shards, self.num_fields], np.int64)
-        bytes_per_field = np.array(list(map(len, self.fields)), np.int64)
-        arrays = header, self.samples_per_shard, self.bytes_per_shard, self.bytes_per_sample, bytes_per_field
-        arrays = np.concatenate(arrays, dtype=np.int64).tobytes()
-        fields = b''.join(map(lambda s: s.encode('utf-8'), self.fields))
-        return arrays + fields
+        magic = 0xDA7AD06E
```

Comment:
What is this magic for?

Reply:
Spend the first 4 bytes on a file signature "doge data", a common pattern (https://en.wikipedia.org/wiki/List_of_file_signatures) used to identify the file type outside of filename

Reply:
Ah, neat!